addon/adapters/adapter.js

```diff
@@ -156,6 +173,46 @@ export default Adapter.extend(ImportExportMixin, {
     return null;
   },
 
+  _queryFilter(record, query = {}) {
+    return keys(query).every((key) => {
+      const value = query[key],
+        matches = key.match(/(.*)Id$/),
```

Comment:
yeah I like that way better then what I suggested. Hmm well I think it is a valid use case, say if you want to find the user that made post `123` without first doing `findRecord('post', '123')`. Also if  you can pass it a regexp, it greatly outpowers `find()`.
```
// get users who've contributed to project.id = 123
query('user', filter: { projects: { id: '123' } });
```
might be a better way of looking at it.


What do you think, do we need arrays to signify that they match two cases on a hasMany?
```
// get users with cats and dogs
query('user', filter: { pets: [{ type: 'cat' }, { type: 'dog' }] });
```